public/JSONtoHTML.coffee

```diff
@@ -0,0 +1,211 @@
+      output += '<div id="' + module.name + '_' + Mclass.name + '_Hover" class="hoverableAnchors"><a href="#' + module.name + '_' + Mclass.name + '" class="fa fa-anchor"></a>&nbsp;&nbsp;&nbsp;&nbsp;<a href="" class="fa fa-file-code"></a></div></div>'
+     
+      if Mclass.documentation != null
+        mdtohtml = marked(Mclass.documentation)
+        output += '<div class="classDocumentation">' + mdtohtml + '</div>'
+
+      if Mclass.longDocumentation != null && Mclass.longDocumentation != undefined
+        mdtohtml = marked(Mclass.longDocumentation)
+        output += '<div class="classLongDocumentation">' + mdtohtml + '</div>'  
+
+      output += '<div class="module_classes_class_methods">'
+      k = 0
+      printedInfo = false
+      while k < Mclass.methods.length
+        Mmethod = Mclass.methods[k]
+
+        if Mmethod.type != null && Mmethod.type != undefined
+          output += '<div class="methodTypeDiv"><p class="MethodTypeHead">' + Mmethod.type + '</p>'
+          
+          ###
+          For method section info, now commented out
+          if Mmethod.documentation != null
+            mdtohtml = marked(Mmethod.documentation)
+            output += '<div class="methodTypeDocumentation">' + mdtohtml + '</div>'
+          ###
+          l = 0
+          while l < Mmethod.methods.length
+            MmethodWithType = Mmethod.methods[l]
+            MmethodWithTypeNameAndType = MmethodWithType.name.split(":")
+            output += '<div id="' + module.name + '_' + Mclass.name + '_' + MmethodWithTypeNameAndType[0].replace(/\s/g,'') + '" class="methodDiv">'
+            output += '<div style="width:100%" onMouseOver="document.getElementById(\'' + module.name + '_' + Mclass.name + '_' + MmethodWithTypeNameAndType[0].replace(/\s/g,'') + '_Hover\').style.display = \'inline-block\'" onMouseOut="document.getElementById(\'' + module.name + '_' + Mclass.name + '_' + MmethodWithTypeNameAndType[0].replace(/\s/g,'') + '_Hover\').style.display = \'none\'">'
+            output += '<p class="methodName">' + MmethodWithTypeNameAndType[0].replace(/\s/g,'')
+            if MmethodWithTypeNameAndType[1] != undefined
+              output += '<code>&nbsp;:' + MmethodWithTypeNameAndType[1] + '</code>'
+            output += '</p>'
+            if MmethodWithType.tag != null && MmethodWithType.tag != undefined 
+              tagType = MmethodWithType.tag.split(":")
+              output += '<div class="tag tag_' + tagType[0].replace(/\s/g,'') + '" style="bottom:0"><p class="<!--fa fa-tags-->">' + tagType[0].replace(/\s/g,'') + ' in ' + tagType[1].replace(/\s/g,'')+ '</p></div>'
+            output += '<div id="' + module.name + '_' + Mclass.name + '_' + MmethodWithTypeNameAndType[0].replace(/\s/g,'') + '_Hover" class="hoverableAnchors"><a href="#' + module.name + '_' + Mclass.name + '_' + MmethodWithTypeNameAndType[0].replace(/\s/g,'') + '" class="fa fa-anchor"></a>&nbsp;&nbsp;&nbsp;&nbsp;<a href="" class="fa fa-file-code"></a></div></div>'
```

Comment:
Yuo're hardcoding a lot of stuff - like that you use `_` to concat part of the names. What if we want to change it to other character? It would be damn hard to refactor. Such things like this separator should always be refactored to a variable. (I suspect that JS have some HTML template engines that make it easier)